src/algebra/char_p/exp_char.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2021 Jakob Scholbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Jakob Scholbach.
+-/
+import algebra.char_p.basic
+import algebra.char_zero
+import data.nat.prime
+
+/-!
+# Exponential characteristic
+
+This file defines the exponential characteristic of an integral domain, 
+and establishes a few basic facts relating the exponential characteristic to the (ordinary)
+characteristic.
+
+## Main results
+- `exp_char` is the definition  
+- `exp_char_is_prime_or_one` the exponential characteristic is a prime or one
+- `char_eq_exp_char_iff` the characteristic equals the exponential characteristic iff the 
+characteristic is prime
+
+## Tags
+exponential characteristic, characteristic
+-/
+
+universe u
+variables (R : Type u) [integral_domain R]
+
+/-- The definition of the exponential characteristic of an integral domain. -/
+class exp_char (q : ℕ) : Prop :=
```

Comment:
I think the definition (but not the results) makes sense for any (comm?)(semi?)ring. It is useful to state this with weakest assumptions possible, because in practice you might want to talk about `exp_char R q` before you have proven that `R` is an integral domain.

Reply:
I have defined it for a semiring now (this seems the maximum generality, given the scope of `char_p`).